pkg/controller/restore_finalizer_controller.go

```diff
@@ -304,6 +306,26 @@ func (ctx *finalizerContext) patchDynamicPVWithVolumeInfo() (errs results.Result
 						return false, err
 					}
 
+					// We are handling a common but specific scenario where a PVC is in a pending state and uses a storage class with
+					// VolumeBindingMode set to WaitForFirstConsumer. In this case, the PV patch step is skipped to avoid
+					// failures due to the PVC not being bound, which could cause a timeout and result in a failed restore.
+					if pvc != nil && pvc.Status.Phase == v1.ClaimPending {
+						// check if storage class used has VolumeBindingMode as WaitForFirstConsumer
+						scName := *pvc.Spec.StorageClassName
+						sc := &storagev1api.StorageClass{}
+						err = ctx.crClient.Get(context.Background(), client.ObjectKey{Name: scName}, sc)
+
+						if err != nil {
+							errs.Add(restoredNamespace, err)
+						}
```

Comment:
Do you want to return after this error occurred?
Without return, it would still try to patch since sc was not retrievable and `*sc.VolumeBindingMode` would be undefined.

Reply:
Updated PR, thanks @kaovilai